### PR TITLE
Initial code import

### PR DESCRIPTION
**Pull request by aleventhal**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,63 @@
-# automation inspector
+# Automation Inspector
+An inspection tool for Chrome Automation API
+
+The [Chrome Automation API](https://developer.chrome.com/extensions/automation)
```
Comment (minorninth) [inline README.md:4]: Maybe mention how it relates to other accessibility APIs on other platforms,
or how it's basically a way to access Chrome accessibility.

File: js/client/props-tree.js [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+      'Table': [ 'tableRowCount', 'tableColumnCount', 'tableCellColumnIndex',
+        'tableCellColumnSpan', 'tableCellRowIndex', 'tableCellRowSpan',
+        'tableColumnHeader', 'tableRowHeader' ],
+      'Live region': [ 'liveStatus', 'liveRelevant', 'liveAtomic', 'liveBusy',
+        'containerLiveStatus', 'containerLiveRelevant', 'containerLiveAtomic',
+        'containerLiveBusy' ],
+      'Selection': [ 'anchorOffset', 'anchorAffinity', 'anchorObject',
+        'focusOffset', 'focusAffinity', 'focusObject', 'textSelStart',
+        'textSelEnd' ],
+      'Text': [ 'language', 'wordStarts', 'wordEnds',
```
Comment (minorninth) [inline js/client/props-tree.js:91]: FYI, wordStarts and wordEnds are only on inline text boxes, whereas
language is on any element, and the markers I think are only on
editable text.

File: js/client/props-tree.js [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+            highlightImpl(changeData[prop], prop + '.');
+          }
+          else {
+            viewNode.addClass('highlight');
+            // Show what the old property value was
+            viewNode.setTitle(viewNode.title +
+              '<span class="was"> (was ' + change + ')</span>');
+            if (fadeMs) {
+              setTimeout(() => {
+                viewNode.removeClass('highlight');
```
Comment (minorninth) [inline js/client/props-tree.js:323]: How do you want to make this accessible? A timeout isn't
really good for accessibility

File: js/client/tree.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+  * Base class for a view tree on the client, extended by
+  * NodeTree, EventTree and PropsTree
+  */
+
+// TODO - perf work on
```
Comment (minorninth) [inline js/client/tree.js:23]: Maybe later migrate these to git issues?

File: js/client/tree.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+// TODO - P4: all searches should be done by background script, passing back
+//            keys, perhaps using alterations to the automation API
+// TODO - P4: allow JSON to be constructed via options dialog
+// TODO - P4: put all config options in JSON
+// Qs:
+//   - htmlTag not working
+//   - Why isn't source=user happening for every click event
+//   - textChanged event, why both event and treeChange?
+//   - treeChanged event, how is this different from other tree changes?
+//   - getImageData() crashes on some nodes
```
Comment (minorninth) [inline js/client/tree.js:51]: Is this still happening? Can you tell me how to repro?

File: js/client/tree.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+//   - Hit inspect button, show same old inspector (ext)
+//--
+//   - DCHECK failing in ax_enums.cc at end of ToString(AXState)
+//   - In app, for event listening, should we limit it to one window at a time?
```
Comment (minorninth) [inline js/client/tree.js]: That would be cool!

File: js/client/tree.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+//     Have a window chooser?
```
Comment (minorninth) [inline js/client/tree.js]: Yes and a hit test function too, I've been implementing full hit test
support in automation recently and it should be ready now

File: js/client/tree.js [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+//   - domQuerySelector -> domQuerySelectorAll
+//   - Window hierarchy contents repeat what's in rootWebView
+//   - How to draw bounds rects, get element under mouse -- contentscript?
+//   - Why no events on paragraph when I change html attribute?
```
Comment (minorninth) [inline js/client/tree.js]: We don't fire events if it's not an accessibility attribute

**Review by minorninth — APPROVED** [review]
Everything looks good.

The style is really good and consistent. I didn't
mention anything style-related that differs from our
usual JS style because this is a standalone project
and it's quite consistent. Did you happen to use any
js linting tools or is any of that done automatically
when you run bower (is that was eslint:recommended
does)?

File: README.md (hunk 1 above)
Comment (aleventhal) [reply README.md:4]: Will do

File: js/client/props-tree.js (hunk 3 above)
Comment (aleventhal) [reply js/client/props-tree.js:323]: It's really an alternative view of the live event stream. If you arrow through the events all the data is presented. I do think that some kind of description of what fields are changed as you do that could be helpful. Ultimately I plan to watch different users using the tool and improve it for what they really need.

File: js/client/tree.js (hunk 4 above)
Comment (aleventhal) [reply js/client/tree.js:23]: Ok

File: js/client/tree.js (hunk 5 above)
Comment (aleventhal) [reply js/client/tree.js:51]: I'll do that as follow-up as I migrate stuff to git/chrome issues.

File: js/client/props-tree.js (hunk 2 above)
Comment (aleventhal) [reply js/client/props-tree.js:91]: Ok, it only shows the fields that are present on any given object anyway.

**Pull request merged**